python/kfserving/kfserving/kfmodel.py

```diff
@@ -52,15 +53,27 @@ def load(self) -> bool:
 
     def preprocess(self, request: Dict) -> Dict:
         # If cloudevent dict, then parse 'data' field. Otherwise, pass through.
-        if  "data" in request \
-            and "time" in request \
-            and "type" in request \
-            and "source" in request \
-            and "id" in request \
-            and "specversion" in request:
-            return request["data"]
-        else:
-            return request
+        response = request
+
+        if(isinstance(request, CloudEvent)):
+            response = request.data
+            if(isinstance(response, bytes)):
+                try:
+                    response = json.loads(response.decode('UTF-8'))
+                except (json.JSONDecodeError):
+                    pass
```

Comment:
I think we should still throw exception if the content type is set to `application/json` but not able to decode, if the content type is not set then we can pass.

Reply:
Done